numpy/core/tests/test_multiarray.py

```diff
@@ -844,7 +844,29 @@ def test_from_string(self):
 
     def test_void(self):
         arr = np.array([], dtype='V')
-        assert_equal(arr.dtype.kind, 'V')
+        assert arr.dtype == 'V8'  # current default
+        # Check that promotion of different length works:
+        arr = np.array([b"1234", b"12345"], dtype="V")
+        assert arr.dtype == 'V5'
+        arr = np.array([b"12345", b"1234"], dtype="V")
+        assert arr.dtype == 'V5'
+        # Check the same for the casting path:
+        arr = np.array([b"1234", b"12345"], dtype="O").astype("V")
```

Comment:
What does this do? Zero-pad? Undefined-data-pad?

Can you add a test for that?